src/components/select/multi-select/multi-select.test.tsx

```diff
@@ -46,60 +67,1111 @@ test("initially renders combobox with custom placeholder when placeholder prop i
+    render(
+      <MultiSelect label="Colour" onChange={() => {}} readOnly>
+        <Option text="amber" value="amber" />
+      </MultiSelect>
+    );
+
+    await user.click(screen.getByRole("textbox"));
+
+    expect(screen.queryByRole("listbox")).not.toBeInTheDocument();
+  });
+
+  it("does not open, when input is read-only and Space bar is pressed", async () => {
+    const user = userEvent.setup();
+    render(
+      <MultiSelect label="Colour" onChange={() => {}} readOnly>
+        <Option text="amber" value="amber" />
+      </MultiSelect>
+    );
+
+    await user.tab();
+    await user.keyboard("[Space]");
+
+    expect(screen.queryByRole("listbox")).not.toBeInTheDocument();
+  });
+
+  it("opens when input is focused and openOnFocus prop is true", async () => {
+    const user = userEvent.setup();
+    render(
+      <MultiSelect label="Colour" onChange={() => {}} openOnFocus>
+        <Option text="amber" value="amber" />
+      </MultiSelect>
+    );
+
+    await user.tab();
+
+    expect(await screen.findByRole("listbox")).toBeVisible();
+  });
+
+  it("does not open, when input is focused and openOnFocus prop is false", async () => {
+    const user = userEvent.setup();
+    render(
+      <MultiSelect label="Colour" onChange={() => {}} openOnFocus={false}>
+        <Option text="amber" value="amber" />
+      </MultiSelect>
+    );
+
+    await user.tab();
+
+    expect(screen.queryByRole("listbox")).not.toBeInTheDocument();
+  });
+
+  it("stays open after an option is clicked", async () => {
+    const user = userEvent.setup();
+    render(
+      <MultiSelect label="Colour" onChange={() => {}}>
+        <Option text="amber" value="amber" />
+      </MultiSelect>
+    );
+
+    await user.click(screen.getByRole("combobox"));
+    await user.click(await screen.findByRole("option", { name: "amber" }));
+
+    expect(screen.getByRole("listbox")).toBeVisible();
+  });
+
+  it("stays open after an option is focused and Enter key is pressed", async () => {
+    const user = userEvent.setup();
+    render(
+      <MultiSelect label="Colour" onChange={() => {}}>
+        <Option text="amber" value="amber" />
+      </MultiSelect>
+    );
+
+    await user.click(screen.getByRole("combobox"));
+    await user.keyboard("ArrowDown");
+    await user.keyboard("Enter");
+
+    expect(screen.getByRole("listbox")).toBeVisible();
+  });
+
+  it("closes when clicking outside the list", async () => {
+    const user = userEvent.setup();
+    render(
+      <>
+        <MultiSelect label="Colour" onChange={() => {}}>
+          <Option text="amber" value="amber" />
+        </MultiSelect>
+        <p>Outside content</p>
+      </>
+    );
+
+    await user.click(screen.getByRole("combobox"));
+    await user.click(screen.getByText("Outside content"));
+
+    await waitFor(() =>
+      expect(screen.queryByRole("listbox")).not.toBeInTheDocument()
+    );
+  });
+
+  it("closes when input is clicked twice", async () => {
```

Comment:
suggestion (non-blocking): might be worth doing the same test for the input icon